scipy/stats/_continuous_distns.py

```diff
@@ -4243,11 +4237,10 @@ def _check_fit_input_parameters(self, data, args, kwds):
                 kwds[key] = val
 
     # determine if there are any unknown arguments in kwds
-    keys = [x for x in kwds.keys() if x not in ['loc', 'scale', 'optimizer',
-                                                'floc', 'fscale',
-                                                *fshape_keys]]
-    if keys:
-        raise TypeError(f"Unknown keyword arguments: {keys}.")
+    known_keys = {'loc', 'scale', 'optimizer', 'floc', 'fscale', *fshape_keys}
+    unknown_keys = set(kwds.keys()).difference(known_keys)
```

Comment:
```suggestion
    unknown_keys = set(kwds).difference(known_keys)
```
I forgot you don't even need to specify `keys`. Not a big deal, though. Just a thought.
And yes, this is a fine way of doing it - and the right way if you want to print the unknown keys. The main point was to use set functionality here. Even if, for some reason, you had to do:
```
[x for x in kwds.keys() if x not in ['loc', 'scale', 'optimizer', 'floc', 'fscale', *fshape_keys]]
```
Then the list should be a set. The speed doesn't matter for something this small, but still - sets are just the right data structure when all you want to do is check membership.